lib/improver/utilities/temporal.py

```diff
@@ -628,6 +629,7 @@ def process(self, cube_t0, cube_t1):
         interpolated_cube = cube.interpolate(time_list, iris.analysis.Linear())
         interpolated_cubes = iris.cube.CubeList()
         for single_time in interpolated_cube.slices_over('time'):
+            check_cube_not_float64(single_time, fix=True)
```

Comment:
Why is this needed?

Reply:
Because iris.analysis converts forecast_period back to float64 and then sav_netcdf will not work as the cube is now float64.